lib/libc/picolibc/CMakeLists.txt

```diff
@@ -13,7 +13,7 @@ if(NOT CONFIG_PICOLIBC_USE_MODULE)
 
   zephyr_compile_options(--specs=picolibc.specs)
   zephyr_compile_definitions(_POSIX_C_SOURCE=200809)
-  zephyr_link_libraries(-T/dev/null --specs=picolibc.specs c -lgcc)
+  zephyr_late_link_libraries(-T/dev/null --specs=picolibc.specs c -lgcc)
```

Comment:
this could be one of those situation where `link_libraries()` is actually acceptable to use.
https://cmake.org/cmake/help/latest/command/link_libraries.html

as we actually want to ensure `-lc` and `-lgcc` are linked last.
But using `link_libraries()` at this location will not work, cause it needs to be on a higher level.

Moving this to higher level would require some changes to how library infrastructure is sourced in the build system.
Some libc impl also creates a library which must be linked, whereas this purely set flags.